cmd/snap/cmd_interface.go

```diff
@@ -41,7 +41,7 @@ type cmdInterface struct {
 	} `positional-args:"true"`
 }
 
-var shortInterfaceHelp = i18n.G("List snap interfaces")
+var shortInterfaceHelp = i18n.G("Show details of snap interfaces")
```

Comment:
It's not for a snap. For example, having `http` installed does not let you do `snap interface http`.

Reply:
without arguments, it returns a list of all the interfaces snapd knows about:
```
$ snap interface
Name                    Summary
avahi-observe           allows discovery on a local network via the mDNS/DNS-SD protocol suite
browser-support         allows access to various APIs needed by modern web browsers
camera                  allows access to all cameras
cups-control            allows access to the CUPS control socket
desktop                 allows access to basic graphical desktop resources
desktop-legacy          allows privileged access to desktop legacy methods
gsettings               allows access to any gsettings item of current user
home                    allows access to non-hidden files in the home directory
joystick                allows access to joystick devices
lxd-support             allows operating as the LXD service
network                 allows access to the network
network-bind            allows operating as a network service
opengl                  allows access to OpenGL stack
optical-drive           allows access to optical drives
pulseaudio              allows operating as or interacting with the pulseaudio service
removable-media         allows access to mounted removable storage
screen-inhibit-control  allows inhibiting the screen saver
snapd-control           allows communicating with snapd
system-observe          allows observing all processes and drivers
unity7                  allows interacting with Unity 7 services
upower-observe          allows operating as or reading from the UPower service
wayland                 allows access to compositors supporting wayland protocol
x11                     allows interacting with or running as an X11 server
```
then you give it one of those names as an argument, and it tells you about that:
```
snap interface x11
name:    x11
summary: allows interacting with or running as an X11 server
plugs:
  - cavestory
  - firefox
  - gimp
  - minecraft
  - mosaic
  - shattered-pixel-dungeon
  - spotify
  - tmnationsforever
  - vlc
  - xbill-xaw
  - xonotic
slots:
  - core
```

Reply:
thanks for the explanation - yes, that makes complete sense.

Reply:
To be precise, it shows the list of interfaces that have some connections. Otherwise the list would be much longer and it is indeed one we show with `snap interface --all`